code/game/objects/items/weapons/melee/melee_misc.dm

```diff
@@ -45,6 +45,368 @@
 	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.5, _parryable_attack_types = NON_PROJECTILE_ATTACKS)
 	RegisterSignal(src, COMSIG_PARENT_QDELETING, PROC_REF(alert_admins_on_destroy))
 
+//Security Sword
+/obj/item/melee/secsword
+	name = "securiblade"
+	desc = "A simple, practical blade developed by Shellguard munitions for ‘enhanced’ riot control."
+	lefthand_file = 'icons/mob/inhands/weapons_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons_righthand.dmi'
+	icon_state = "secsword0"
+	item_state = "secsword0"
+	flags = CONDUCT
+	force = 15
+	throwforce = 5
+	w_class = WEIGHT_CLASS_BULKY
+	sharp = TRUE
+	origin_tech = "combat=4"
+	attack_verb = list("stabbed", "slashed", "sliced")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	materials = list(MAT_METAL = 1000)
+	needs_permit = TRUE
+
+	///Mode Defines
+	#define SECSWORD_OFF 0
+	#define SECSWORD_STUN 1
+	#define SECSWORD_BURN 2
+
+	var/base_icon = "secsword"
+	var/stam_damage = 35 // 3 hits to stamcrit
+	var/burn_damage = 10
+	/// How much power does it cost to stun someone
+	var/stam_hitcost = 1000
+	/// How much power does it cost to burn someone
+	var/burn_hitcost = 1500
+	/// the initial cooldown tracks the time between stamina damage. tracks the world.time when the baton is usable again.
+	var/cooldown = 3.5 SECONDS
+	var/obj/item/stock_parts/cell/high/cell = null
```

Comment:
Does this need to be this specifically typed? I think you might want this to be a more generic type.

Reply:
This should be typed to be a high cell, yes. The reason being that it is the type of cell the sword starts with. Future cells are accepted just fine in testing - this is mostly for the init link_new_cell(). Otherwise, it would be a var/cell .